### PR TITLE
chore: remove clang-format-7 references

### PR DESCRIPTION
**Pull request by themarwhal**
## Summary
Remove references to clang-format-7 in dev env setup config / scripts

## Test Plan
destroyed and provisioned all VMs
CI check to make sure devcontainer builds
Ran `magma git:(remove-clang-format-7) ✗ docker build -f lte/gateway/docker/mme/Dockerfile.ubuntu18.04 .`

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by uri200 — APPROVED** [review]

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -85,7 +84,7 @@ RUN echo "Install general purpose packages" && \
         gem install fpm && \
         update-alternatives --install /usr/bin/clang clang /usr/lib/llvm-11/bin/clang 10 && \
         update-alternatives --install /usr/bin/clang++ clang++ /usr/lib/llvm-11/bin/clang++ 10 && \
-        update-alternatives --install /usr/bin/clang-format clang-format /usr/lib/llvm-7/bin/clang-format 10 && \
+        update-alternatives --install /usr/bin/clang-format clang-format /usr/lib/llvm-7/bin/clang-format-11 10 && \
```
Comment (Neudrino) [inline .devcontainer/Dockerfile]:
```suggestion
        update-alternatives --install /usr/bin/clang-format clang-format /usr/lib/llvm-11/bin/clang-format-11 10 && \
```
:question:

**Comment by Neudrino** [comment]
Would we also want to remove it from https://github.com/magma/magma/blob/master/lte/gateway/docker/mme/Dockerfile.ubuntu18.04#L22? Or is it kept for backwards compatibility there?

**Comment by Neudrino** [comment]
Maybe a naive question:
Does the change of https://github.com/magma/magma/blob/master/lte/gateway/deploy/roles/dev_common/tasks/main.yml affect the deployability / bugfixing / backporting of v1.6?

**Comment by themarwhal** [comment]
> Maybe a naive question:
> Does the change of https://github.com/magma/magma/blob/master/lte/gateway/deploy/roles/dev_common/tasks/main.yml affect the deployability / bugfixing / backporting of v1.6?

I think we are ok there. Because of the formatting change, backporting will be a pain regardless. And hopefully we do not have any back ports after we release 1.7.

**Comment by themarwhal** [comment]
> Maybe a naive question:
> Does the change of https://github.com/magma/magma/blob/master/lte/gateway/deploy/roles/dev_common/tasks/main.yml affect the deployability / bugfixing / backporting of v1.6?

I tried, but I was getting some errors. I'm wondering if clang-format-11 is not available on ubuntu 18? Maybe we can update the base image to ubuntu 20 for this one. (cc @rdefosse) I'm also not seeing how clang-format is used for the dockerfile, so maybe we don't need it to begin with.

**Comment by Neudrino** [comment]
> I'm wondering if clang-format-11 is not available on ubuntu 18?

The newest version available in Bionic Beaver is
`clang-format-10/bionic-updates,bionic-security 1:10.0.0-4ubuntu1~18.04.2 amd64`, afaiks.
Obtained by `apt search clang-format` in a pure Bionic docker image.

**Comment by themarwhal** [comment]
> > I'm wondering if clang-format-11 is not available on ubuntu 18?
> 
> The newest version available in Bionic Beaver is `clang-format-10/bionic-updates,bionic-security 1:10.0.0-4ubuntu1~18.04.2 amd64`, afaiks. Obtained by `apt search clang-format` in a pure Bionic docker image.

ah ok! let me try that

**Comment by Neudrino** [comment]
> Maybe we can update the base image to ubuntu 20 for this one.

`clang-format-11` as changed in https://github.com/magma/magma/commit/0c1d089f2a20af1c8863b7301cb732f1f1510020 would be available in Ubuntu focal.
However, it seems a dist-upgrade is not as easy, as I get an error installing third party dependecies on Focal Fossa:
```
Package nlohmann-json-dev is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

E: Package 'nlohmann-json-dev' has no installation candidate
```

**Comment by Neudrino** [comment]
Overall, the topic of `lte/gateway/docker/mme/Dockerfile.ubuntu18.04` might be out of scope of this PR and worth a separate issue?
Maybe one can upgrade to focal in a separate effort, and leave it as is until than?

**Comment by themarwhal** [comment]
> Overall, the topic of `lte/gateway/docker/mme/Dockerfile.ubuntu18.04` might be out of scope of this PR and worth a separate issue?
> Maybe one can upgrade to focal in a separate effort, and leave it as is until than?

I like this idea :D

**Comment by Neudrino** [comment]
Overall, the topic of `lte/gateway/docker/mme/Dockerfile.ubuntu18.04` might be out of scope of this PR and worth a separate issue?
Maybe one can upgrade to focal in a separate effort, and leave it as is until than?

> > Overall, the topic of `lte/gateway/docker/mme/Dockerfile.ubuntu18.04` might be out of scope of this PR and worth a separate issue?
> > Maybe one can upgrade to focal in a separate effort, and leave it as is until than?
> 
> I like this idea :D

I created issue https://github.com/magma/magma/issues/11608 for it. Feel free to refine if needed.

**Pull request merged**